site/site.sh

```diff
@@ -3,4 +3,23 @@ set -ex
 
 /bin/bash /poll.sh &
 
+rewrite_conf_location=/etc/nginx/conf.d/rewrite-links-for-namespace.conf
+
+if [ $HAIL_DEFAULT_NAMESPACE == "default" ]
+then
+    prefix=""
+    cat >$rewrite_conf_location <<EOF
+# no rewriting necessary because we are in the default namespace
+EOF
+else
+    prefix="/$HAIL_DEFAULT_NAMESPACE/site"
+    cat >$rewrite_conf_location <<EOF
+subs_filter href="/([^/]) href="/$HAIL_DEFAULT_NAMESPACE/site/\$1 gr;
+subs_filter src="/([^/]) src="/$HAIL_DEFAULT_NAMESPACE/site/\$1 gr;
```

Comment:
Rewriting HTML with regexes always felt a little wrong to me, but I don't have a better suggestion and if it works...

Having a non-standard root seems like such an anti-pattern.